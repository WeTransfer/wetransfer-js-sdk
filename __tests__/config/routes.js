const routes = require('../../src/config/routes');

describe('Routes configuration', () => {
  it('should define the right prefix', async () => {
    expect(routes.prefix).toBe('/v1');
  });

  it('should define an authorize path', () => {
    expect(routes.authorize).toEqual({ url: '/v1/authorize' });
  });

  it('should define a transfers path', () => {
    expect(routes.transfers).toEqual({ url: '/v1/transfers' });
  });

  it('should define an items path', () => {
    expect(routes.items('id')).toEqual({ url: '/v1/transfers/id/items' });
  });

  it('should define a multipart upload path', () => {
    expect(routes.multipart({
      id: 'item_id',
      meta: {
        multipart_upload_id: 'multipart_upload_id'
      }
    }, 1)).toEqual({ url: '/v1/files/item_id/uploads/1/multipart_upload_id', method: 'get' });
  });

  it('should define an upload path', () => {
    expect(routes.upload('https://dev.wetransfer.com/very-long-url')).toEqual({ url: 'https://dev.wetransfer.com/very-long-url' });
  });

  it('should define an uploadComplete path', () => {
    expect(routes.uploadComplete('id')).toEqual({ url: '/v1/files/id/uploads/complete' });
  });
});
