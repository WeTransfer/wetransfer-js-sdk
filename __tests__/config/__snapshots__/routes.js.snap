// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes configuration boards namespace should define a find path 1`] = `
Object {
  "method": "get",
  "url": "/v2/boards/board-id",
}
`;

exports[`Routes configuration boards namespace should define a multipart upload path 1`] = `
Object {
  "method": "get",
  "url": "/v2/boards/board-id/files/file-id/upload-url/1/multipart-upload-id",
}
`;

exports[`Routes configuration boards namespace should define all endpoints 1`] = `
Object {
  "addFiles": [Function],
  "addLinks": [Function],
  "create": Object {
    "url": "/v2/boards",
  },
  "find": [Function],
  "multipart": [Function],
  "upload": [Function],
  "uploadComplete": [Function],
}
`;

exports[`Routes configuration boards namespace should define an addFiles path 1`] = `
Object {
  "url": "/v2/boards/board-id/files",
}
`;

exports[`Routes configuration boards namespace should define an addLinks path 1`] = `
Object {
  "url": "/v2/boards/board-id/links",
}
`;

exports[`Routes configuration boards namespace should define an upload path 1`] = `
Object {
  "url": "s3://upload-url",
}
`;

exports[`Routes configuration boards namespace should define an uploadComplete path 1`] = `
Object {
  "method": "put",
  "url": "/v2/boards/board-id/files/file-id/upload-complete",
}
`;

exports[`Routes configuration transfers namespace should define a find path 1`] = `
Object {
  "method": "get",
  "url": "/v2/transfers/transfer-id",
}
`;

exports[`Routes configuration transfers namespace should define a multipart upload path 1`] = `
Object {
  "method": "get",
  "url": "/v2/transfers/transfer-id/files/file-id/upload-url/1",
}
`;

exports[`Routes configuration transfers namespace should define all endpoints 1`] = `
Object {
  "create": Object {
    "url": "/v2/transfers",
  },
  "finalize": [Function],
  "find": [Function],
  "multipart": [Function],
  "upload": [Function],
  "uploadComplete": [Function],
}
`;

exports[`Routes configuration transfers namespace should define an finalize path 1`] = `
Object {
  "method": "put",
  "url": "/v2/transfers/transfer-id/finalize",
}
`;

exports[`Routes configuration transfers namespace should define an upload path 1`] = `
Object {
  "url": "s3://upload-url",
}
`;

exports[`Routes configuration transfers namespace should define an uploadComplete path 1`] = `
Object {
  "method": "put",
  "url": "/v2/transfers/transfer-id/files/file-id/upload-complete",
}
`;
